### PR TITLE
Instrumentation: Log plugin and datasource info for each plugin request. 

### PR DESCRIPTION
**Pull request by bergquist**
Log each datasource requested proxied to core or plugin datasources. These logs can be used to catch errors for certain types or datasources. 

The log also contains the traceId if the request is traced to enable further investigations.

### CODE REVIEW

File: pkg/plugins/backendplugin/instrumentation/instrumentation.go [hunk 1]
```diff
@@ -34,29 +42,51 @@ func instrumentPluginRequest(pluginID string, endpoint string, fn func() error)
...
+			logParams = append(logParams, "traceID", traceID)
+		}
+
+		if pluginCtx.DataSourceInstanceSettings != nil {
+			logParams = append(logParams, "ds_name", pluginCtx.DataSourceInstanceSettings.Name)
+			logParams = append(logParams, "ds_uid", pluginCtx.DataSourceInstanceSettings.UID)
+			logParams = append(logParams, "ds_type", pluginCtx.DataSourceInstanceSettings.Type)
+		}
+
+		logger.Info("Plugin Request Completed", logParams...)
```
Comment (marefr) [inline pkg/plugins/backendplugin/instrumentation/instrumentation.go:68]: Debug?

```suggestion
		logger.Debug("Plugin Request Completed", logParams...)
```

File: pkg/plugins/backendplugin/instrumentation/instrumentation.go [hunk 2]
```diff
@@ -34,29 +42,51 @@ func instrumentPluginRequest(pluginID string, endpoint string, fn func() error)
...
+
+		traceID := tracing.TraceIDFromContext(ctx, false)
+		if traceID != "" {
+			logParams = append(logParams, "traceID", traceID)
+		}
+
+		if pluginCtx.DataSourceInstanceSettings != nil {
+			logParams = append(logParams, "ds_name", pluginCtx.DataSourceInstanceSettings.Name)
+			logParams = append(logParams, "ds_uid", pluginCtx.DataSourceInstanceSettings.UID)
+			logParams = append(logParams, "ds_type", pluginCtx.DataSourceInstanceSettings.Type)
```
Comment (marefr) [inline pkg/plugins/backendplugin/instrumentation/instrumentation.go:65]: You're mixing snake_case and camelCase here. Please try to be consistent. I think most of our logs uses snakeCase.

File: pkg/plugins/manager/client/client.go [hunk 3]
```diff
@@ -11,17 +11,20 @@ import (
...
 	"github.com/grafana/grafana/pkg/plugins/backendplugin/instrumentation"
 	"github.com/grafana/grafana/pkg/plugins/manager/registry"
+	"github.com/grafana/grafana/pkg/setting"
 )
 
 var _ plugins.Client = (*Service)(nil)
 
 type Service struct {
 	pluginRegistry registry.Service
+	cfg            *setting.Cfg
```
Comment (wbrowne) [inline pkg/plugins/manager/client/client.go:21]: If it's not too much trouble, would you mind using `*plugins.Cfg` as a dependency here? Feel free to add a new `bool` field for the feature toggle state. Then the `ProvideService` can turn the `*setting.Cfg` into the right thing with `plugins.FromGrafanaCfg`. It's our way of strangling the plugins package as much as possible 😄

File: pkg/plugins/backendplugin/instrumentation/instrumentation.go (hunk 2 above)
Comment (wbrowne) [inline pkg/plugins/backendplugin/instrumentation/instrumentation.go:65]: Shouldn't need `Type` as it's the same as `PluginContext.PluginId`

File: pkg/plugins/backendplugin/instrumentation/instrumentation.go (hunk 1 above)
Comment (bergquist) [reply pkg/plugins/backendplugin/instrumentation/instrumentation.go:68]: Nah we want these logs for every plugin request in prod. turning on debug everywhere would be too much :)

This is the first stab at solving https://github.com/grafana/grafana/discussions/43120

File: pkg/plugins/backendplugin/instrumentation/instrumentation.go (hunk 2 above)
Comment (bergquist) [reply pkg/plugins/backendplugin/instrumentation/instrumentation.go:65]: :+1:

File: pkg/plugins/manager/client/client.go (hunk 3 above)
Comment (bergquist) [reply pkg/plugins/manager/client/client.go:21]: :+1:

File: pkg/plugins/backendplugin/instrumentation/instrumentation.go (hunk 1 above)
Comment (bergquist) [reply pkg/plugins/backendplugin/instrumentation/instrumentation.go:67]: Expect more changes behind this feature flag :)

Comment (marefr) [reply pkg/plugins/backendplugin/instrumentation/instrumentation.go:67]: Okay, but whenever you remove the feature toggle please change to debug or a setting ;)

Comment (bergquist) [reply pkg/plugins/backendplugin/instrumentation/instrumentation.go:67]: > a setting ;)

that would be fine. Similar to the request logger.

**Comment by marefr** [comment]
@bergquist let us know if you need any help with this one

**Comment by bergquist** [comment]
@marefr should be :)

File: pkg/plugins/backendplugin/instrumentation/instrumentation.go [hunk 4]
```diff
@@ -34,29 +41,50 @@ func instrumentPluginRequest(pluginID string, endpoint string, fn func() error)
...
+
+	if cfg.LogPluginRequests {
+		logParams := []interface{}{
+			"status", status,
+			"duration", elapsed,
+			"pluginId", pluginCtx.PluginID,
+			"endpoint", endpoint,
+		}
+
+		traceID := tracing.TraceIDFromContext(ctx, false)
```
Comment (marefr) [inline pkg/plugins/backendplugin/instrumentation/instrumentation.go:56]: Nit. Could use the contextual logger to automatically get this set.

**Review by marefr — APPROVED** [review]
LGTM great

File: pkg/plugins/backendplugin/instrumentation/instrumentation.go (hunk 4 above)
Comment (bergquist) [reply pkg/plugins/backendplugin/instrumentation/instrumentation.go:56]: Ah good point. I think we added that after I started this PR :)

**Review by wbrowne — APPROVED** [review]
🚀

**Pull request merged**